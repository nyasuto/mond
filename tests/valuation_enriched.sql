INSERT INTO assets (ticker, ccy) VALUES ('VTI','USD');
INSERT INTO assets (ticker, ccy) VALUES ('TOPIX','JPY');

INSERT INTO fx_rates (date, pair, rate) VALUES ('2023-12-29','USDJPY',140.0);
INSERT INTO fx_rates (date, pair, rate) VALUES ('2023-12-30','USDJPY',142.0);

INSERT INTO snapshots (date, ticker, qty, price_ccy) VALUES ('2023-12-29','VTI',10,200);
INSERT INTO snapshots (date, ticker, qty, price_ccy) VALUES ('2023-12-29','TOPIX',5,100);
INSERT INTO snapshots (date, ticker, qty, price_ccy) VALUES ('2023-12-30','VTI',11,205);
INSERT INTO snapshots (date, ticker, qty, price_ccy) VALUES ('2023-12-30','TOPIX',5,101);

SELECT
  date,
  ticker,
  round(value_jpy, 3) AS value_jpy,
  round(portfolio_value_jpy, 3) AS portfolio_value_jpy,
  round(weight, 6) AS weight
FROM v_valuation_enriched
ORDER BY date, ticker;
