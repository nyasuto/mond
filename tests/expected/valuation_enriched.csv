date,ticker,value_jpy,portfolio_value_jpy,weight
2023-12-29,TOPIX,500.0,280500.0,0.001783
2023-12-29,VTI,280000.0,280500.0,0.998217
2023-12-30,TOPIX,505.0,320715.0,0.001575
2023-12-30,VTI,320210.0,320715.0,0.998425
